lld/test/MachO/objc-category-merging-swift.s

```diff
@@ -0,0 +1,357 @@
+# REQUIRES: aarch64
+# RUN: rm -rf %t; mkdir %t && cd %t
+
+############ Test merging multiple categories into a single category ############
+## Apply category merging to swiftc code just make sure we can handle addends
+## and don't erase cateogry names for swift -- in order to not crash
+# RUN: llvm-mc -filetype=obj -triple=arm64-apple-macos -o cat_swift.o %s
+# RUN: %lld -arch arm64 -dylib -o cat_swift.dylib cat_swift.o -objc_category_merging
```

Comment:
Are you just validating that it doesn't crash? No checks on the output?

Reply:
Yes, the goal is to not crash. But I added a check for the metadata also, for completeness.